src/boot/efi/xbootldr.c

```diff
@@ -12,33 +12,30 @@ union GptHeaderBuffer {
         uint8_t space[CONST_ALIGN_TO(sizeof(EFI_PARTITION_TABLE_HEADER), 512)];
 };
 
-static EFI_DEVICE_PATH *path_chop(EFI_DEVICE_PATH *path, EFI_DEVICE_PATH *node) {
+static EFI_DEVICE_PATH *path_replace(
+                const EFI_DEVICE_PATH *path,
+                const EFI_DEVICE_PATH *node,
+                const EFI_DEVICE_PATH *new_node) {
+
         assert(path);
         assert(node);
```

Comment:
maybe a comment here that `path` and `node` must be part of the same path object, and that `new_node` must be a different object, would be good. Otherwise this is very hard to grok.